### PR TITLE
update kyma-addon to kyma 1.10

### PR DESCRIPTION
**Pull request by a-thaler**
**What this PR does / why we need it**:
Updates the kyma addon to Kyma 1.10 with the following benefits:
- Support for kubernetes 1.16
- Gardener certificate logic removed from addon as it is part of kyma-installer
- Kyma version has improvements in stability

**Which issue(s) this PR fixes**:
Follow-Up on #1754

**Special notes for your reviewer**:
Only GCP and Azure are supported. 

**Release note**:
```improvement operator
Update of the temporary, experimental Kyma addon to latest Kyma version 1.10. It can be installed onto shoot clusters out-of-the-box by annotating the `Shoot` with `experimental.addons.shoot.gardener.cloud/kyma=enabled`. Be aware that we won't provide upgrades or customization, and that this addon is temporary and will be removed in a future version of Gardener again. Its purpose is to ease the Kyma installation and to show-case which features it provides. It is by no means a production-ready setup. Also, please note that, once enabled, the Kyma addon can never be disabled again. The only way to get rid of it is to delete the shoot cluster. You can check the status of the installation by using `kubectl get installation/kyma-installation -o jsonpath="{'Status: '}{.status.state}{', description: '}{.status.description}"`.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
If we would merge this we would automatically roll out this version to existing Kyma installations. AFAIK Kyma is not upgradeable as of today, correct? Or would this work?

**Comment by a-thaler** [comment]
Kyma can be upgraded as long as you do it one by another 1.8.x -> 1.9.x -> 1.10.x
As the upgrade should be triggered conscious (deprecated things might be removed) I prefer to keep the clusters in the version they are, the owners can trigger the update manual at any time.
So a re-apply of the addon will happen on changing the addon version? Are there other triggers for a re-apply?

**Review by vpnachev — COMMENTED** [review]
How is this PR aligned with the release note of #1754 `...Be aware that we won't provide upgrades or customization...`  ?

**Comment by a-thaler** [comment]
@vpnachev Indeed my goal is just to provide a newer base version for new clusters. Otherwise, the addon has no value at all as it will be based on some outdated version.
I don't want that existing installations are getting upgraded automatically, if people want an upgrade they can do it in a manual way.
It was good feedback by @rfranzke that an automatic apply of the templates will happen.
I will add a check to not deploy the kyma-installer in case of an existing installation.

**Comment by a-thaler** [comment]
I assured that a re-apply will work by having the job names unique per version.
I labeled the jobs with the version and added an `if` to the job logic to not execute if there is a different version installed already.
With that, no automatic upgrade will happen but new clusters will get 1.10 version.
Please have another look.

File: charts/shoot-addons-kyma/charts/kyma/values.yaml [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+  resources:
+    limits:
+      cpu: 100m
+      memory: 128Mi
+    requests:
+      cpu: 50m
+      memory: 64Mi
 kyma:
-  version: "1.8.0"
+  version: "1.10.0"
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/values.yaml]: This is used as a job name above. This won't work. Kubernetes object names must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character.

File: charts/shoot-addons-kyma/charts/kyma/files/tiller.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
 set -e
-
-echo "---> Applying Tiller for ${KYMA_VERSION}"
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_VERSION/installation/resources/tiller.yaml
+if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION})" ]; then
+    echo "---> Applying Tiller for ${KYMA_VERSION}"
+    kubectl apply -f "https://raw.githubusercontent.com/kyma-project/kyma/${KYMA_VERSION}/installation/resources/tiller.yaml"
+else
+    echo "---> Skipping Tiller installation as there is a Kyma installation already"
+fi
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/tiller.sh]: missing new line

File: charts/shoot-addons-kyma/charts/kyma/files/installation.sh [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
+    echo "---> Applying Kyma-Installer for ${KYMA_VERSION}"
+    kubectl apply -f "https://raw.githubusercontent.com/kyma-project/kyma/${KYMA_VERSION}/installation/resources/installer.yaml"
 
-echo "---> Apply Kyma installation CR for ${KYMA_VERSION}"
-kubectl -n default apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_VERSION/installation/resources/installer-cr-cluster.yaml.tpl
+    echo "---> Apply Kyma installation CR for ${KYMA_VERSION}"
+    kubectl apply -f "https://raw.githubusercontent.com/kyma-project/kyma/${KYMA_VERSION}/installation/resources/installer-cr-cluster.yaml.tpl"
+else
+    echo "---> Skipping Kyma installation as there is a Kyma installation already"
+fi
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/installation.sh]: missing new line

File: charts/shoot-addons-kyma/charts/kyma/files/installation.sh [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
+if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION})" ]; then
+    echo "---> Applying Kyma-Installer for ${KYMA_VERSION}"
+    kubectl apply -f "https://raw.githubusercontent.com/kyma-project/kyma/${KYMA_VERSION}/installation/resources/installer.yaml"
 
-echo "---> Apply Kyma installation CR for ${KYMA_VERSION}"
-kubectl -n default apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_VERSION/installation/resources/installer-cr-cluster.yaml.tpl
+    echo "---> Apply Kyma installation CR for ${KYMA_VERSION}"
+    kubectl apply -f "https://raw.githubusercontent.com/kyma-project/kyma/${KYMA_VERSION}/installation/resources/installer-cr-cluster.yaml.tpl"
+else
+    echo "---> Skipping Kyma installation as there is a Kyma installation already"
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/installation.sh]:
```suggestion
    echo "---> Skip Kyma installation as there is already a Kyma installation"
```

File: charts/shoot-addons-kyma/charts/kyma/files/installation.sh [hunk 5]
```diff
@@ -1,8 +1,12 @@
 #!/bin/bash
 set -e
 
-echo "---> Applying Kyma-Installer for ${KYMA_VERSION}"
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_VERSION/installation/resources/installer.yaml
+if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION})" ]; then
+    echo "---> Applying Kyma-Installer for ${KYMA_VERSION}"
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/installation.sh]:
```suggestion
    echo "---> Apply Kyma-Installer for ${KYMA_VERSION}"
```

File: charts/shoot-addons-kyma/charts/kyma/files/tiller.sh [hunk 6]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 set -e
-
-echo "---> Applying Tiller for ${KYMA_VERSION}"
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_VERSION/installation/resources/tiller.yaml
+if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION})" ]; then
+    echo "---> Applying Tiller for ${KYMA_VERSION}"
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/tiller.sh]:
```suggestion
    echo "---> Apply Tiller for ${KYMA_VERSION}"
```

File: charts/shoot-addons-kyma/charts/kyma/files/tiller.sh [hunk 7]
```diff
@@ -1,5 +1,8 @@
...
+    kubectl apply -f "https://raw.githubusercontent.com/kyma-project/kyma/${KYMA_VERSION}/installation/resources/tiller.yaml"
+else
+    echo "---> Skipping Tiller installation as there is a Kyma installation already"
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/tiller.sh]:
```suggestion
    echo "---> Skip Tiller installation as there is already a Kyma installation"
```

File: charts/shoot-addons-kyma/charts/kyma/values.yaml [hunk 8]
```diff
@@ -1,12 +1,19 @@
...
   subdomain: "kyma"
 requires:
   k8s:
-    version: "1.15"
+    version: "1.16"
   gardener:
-    extensions: "shoot-cert-service,shoot-dns-service"
+    extensions: "shoot-cert-service,shoot-dns-service"
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/values.yaml]: missing new line

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: charts/shoot-addons-kyma/charts/kyma/files/tiller.sh [hunk 9]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 set -e
-
-echo "---> Applying Tiller for ${KYMA_VERSION}"
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_VERSION/installation/resources/tiller.yaml
+if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION})" ]; then
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/tiller.sh]:
```suggestion
if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION} --ignore-not-found)" ]; then
```

File: charts/shoot-addons-kyma/charts/kyma/files/installation.sh [hunk 10]
```diff
@@ -1,8 +1,12 @@
 #!/bin/bash
 set -e
 
-echo "---> Applying Kyma-Installer for ${KYMA_VERSION}"
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_VERSION/installation/resources/installer.yaml
+if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION})" ]; then
```
Comment (suleymanakbas91) [inline charts/shoot-addons-kyma/charts/kyma/files/installation.sh]:
```suggestion
if [ -z "$(kubectl -n kyma-installer get job -l app=kyma-initializer,version!=${KYMA_VERSION} --ignore-not-found)" ]; then
```

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

**Review by suleymanakbas91 — DISMISSED** [review]

**Comment by rfranzke** [comment]
Puh, I can't overlook this, I can't really judge whether it's doing what it should. I assume you tested that existing Kyma installations are not upgraded and new installations get the new versions. As @suleymanakbas91 approved already and as the addon is experimental only /lgtm from my side as well (without deeply looking into the changes).
Can you please squash your commits so that we can get it merged?

**Review by rfranzke — APPROVED** [review]

**Pull request merged**